### ISSUE TITLE
[High] services/route-matrix-cpp/main.cpp - O(N^2) response amplification + unbounded thread-per-connection DoS

### ISSUE THREAD

**Issue by ionfwsrijan** [KanishJebaMathewM/Truxify#11421]
## Description
`services/route-matrix-cpp/main.cpp` exposes a raw socket server. Two scaling defects:
1. **O(N²) response amplification** — the request body is capped at `MAX_REQUEST_BYTES = 64*1024` (line 32) but `compute_matrix_json` (line 86) emits one JSON object per *ordered pair* of locations. A 64 KB body can carry ~2,000 locations → ~4,000,000 objects (hundreds of MB) built into a `std::stringstream` and `send()`d (line 269).
2. **Unbounded detached threads** — each accepted connection is handled by `std::thread(...).detach()` (line 332) with no pool or limit. With `SO_RCVTIMEO = 30s`, an attacker opens many idle connections (slowloris) → unbounded thread/stack exhaustion and process hang.

## Expected Behavior
Cap the number of locations (and/or the output size) before computing, and serve connections from a bounded thread pool with a max in-flight limit (and a per-request send cap).

## Actual Behavior
A tiny request triggers massive CPU + memory amplification, and a handful of idle connections can exhaust the process. This is a trivial DoS on a network-facing service.

## Proposed Fix
Guard the location count and bound concurrency (multi-line change):

```cpp
if (locs.size() > 512) {
    response = build_response("{\"error\":\"too many locations\"}", "413 Request Entity Too Large");
} else {
    response = build_response(compute_matrix_json(locs), "200 OK");
}
// Bounded pool instead of detach():
static std::mutex pool_mu; static std::condition_variable pool_cv; static int active = 0;
{
    std::unique_lock<std::mutex> lk(pool_mu);
    pool_cv.wait(lk, []{ return active < MAX_CONCURRENT; });
    ++active;
}
std::thread([client]{ handle_client(client); closesocket(client);
    { std::unique_lock<std::mutex> lk(pool_mu); --active; } pool_cv.notify_one(); }).detach();
```
